app/validators/application_fulfilment_validator.rb

```diff
@@ -28,6 +28,12 @@ def perform_validations
       ]
     end
 
+    if record.case.case_type.nil? && record.is_means_tested == 'yes'
```

Comment:
Recommend switching the conditions around:

`if record.is_means_tested == 'yes' && record.case.case_type.nil? ...` so that there is no need to hit the ActiveRecord cache or worse the database